src/PresentationalComponents/Filters/OsVersionFilter.js

```diff
@@ -1,71 +1,44 @@
-
-    let { os: currentValue } = currentFilter;
-    const currentOsVersionsArray = typeof currentValue === 'string' && currentValue.split(',') || [];
-
-    const filterByOsType = (_, value) => {
-        if (currentValue && !currentValue.includes(value)) {
-            apply({ filter: { os: `${currentOsVersionsArray.join(',')},${value}` } });
-        }
-        else if (currentValue && currentValue.includes(value)) {
-            const remainingOs = currentOsVersionsArray.filter(os => os !== value);
-
-            apply({ filter: { os: remainingOs.length > 0 && `${remainingOs.join(',')}` || undefined } });
-        } else {
-            apply({ filter: { os: value !== '' && value || undefined  } });
-        }
-    };
-
-    const onToggle = (isOpen) => {
-        setOpen(isOpen);
-    };
-
-    const onViewMoreClick = () => {
-        setNumOptions(versionFromNewestToOldest.length);
-    };
-
-    return (
+const useOsVersionFilter = (currentFilter = {}, apply) => {
+    const versions =
+        useSelector(({ entities }) => entities?.operatingSystems) || [];
+    const [{ toGroupSelectionValue, buildOSFilterConfig } = {}] = useLoadModule(
         {
-            type: conditionalFilterType.custom,
-            label: intl.formatMessage(messages.labelsFiltersOsVersion),
-            value: 'custom',
-            filterValues: {
-                children: (
-                    <Select
-                        variant={SelectVariant.checkbox}
-                        typeAheadAriaLabel={intl.formatMessage(messages.labelsFiltersOsVersionPlaceholder)}
-                        onToggle={onToggle}
-                        onSelect={filterByOsType}
-                        selections={currentOsVersionsArray}
-                        isOpen={isOpen}
-                        aria-labelledby={'patch-os-version-filter'}
-                        placeholderText={intl.formatMessage(messages.labelsFiltersOsVersionPlaceholder)}
-                        {...(numOptions < versionFromNewestToOldest.length
-                            && { loadingVariant: { text: 'View more', onClick: onViewMoreClick } })}
-                        style={{ maxHeight: '400px', overflow: 'auto' }}
-                    >
-                        {versionFromNewestToOldest.slice(0, numOptions).map((option, index) => (
-                            <SelectOption
-                                key={index}
-                                value={option.value}
-                            />
-                        ))}
-                    </Select>
-                )
-            }
+            appName: 'inventory',
+            scope: 'inventory',
+            module: './OsFilterHelpers'
         }
     );
+
+    let { os: currentValue } = currentFilter;
+    const osVersionValue = (
+        (typeof currentValue === 'string' && currentValue.split(',')) ||
+        []
+    ).map((version) => version.substr(5));
```

Comment:
substr function is [deprecated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr). Use [substring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring) instead.